pcweb/pages/docs/gallery/state.py

```diff
@@ -0,0 +1,109 @@
+import json
+import httpx
+import reflex as rx
+from rxconfig import config
+
+class SideBarState(rx.State):
+    """Side Bar State."""
+
+    community_apps_list: list[dict[str, str]]
+    example_apps_list: list[dict[str, str]]
+
+    page: int = 1
+    sort_by: str = "page_views"
+
+    tags_list: list[str]
+    chosen_tags: set[str]
+
+
+    def set_sort_by(self, sort_by: str):
+        self.sort_by = sort_by
+        self.fetch_apps_list()
+
+    def set_page(self, page: int):
+        if page < 1:
+            page = 1
+        elif page > len(self.community_apps_list):
+            page = len(self.community_apps_list)
```

Comment:
can page number be the len of community apps? or round up of the app number divided by 16?

Reply:
Nice catch fixed